src/Nancy/Security/SSLProxy.cs

```diff
@@ -0,0 +1,42 @@
+﻿namespace Nancy.Security
+{
+    using System;
+    using System.Linq;
+
+    using Nancy.Bootstrapper;
+
+    /// <summary>
+    /// Allows a BeforeRequest hook to change Url to HTTPS if X-Forwarded-Proto header present
+    /// </summary>
+    public class SSLProxy

```

Comment:
You mean `public static class Csrf`? Doesn't mean it shouldn't be static. Why would you create an instance of `SSLProxy`?